logical/system_view.go

```diff
@@ -37,6 +39,17 @@ type SystemView interface {
 
 	// ReplicationState indicates the state of cluster replication
 	ReplicationState() consts.ReplicationState
+
+	// ResponseWrapData wraps the given data in a cubbyhole and returns the
+	// token used to unwrap.
+	ResponseWrapData(data map[string]interface{}, ttl time.Duration, jwt bool) (string, error)
```

Comment:
I haven't done this yet, I think this might be okay for a follow up PR or to be added when it's needed. It would require a change to the core `wrapInCubbyHole` function. What do you think?